### PR TITLE
Herwig7: Updated config files

### PR DESCRIPTION
**Pull request by mharrend**
LHAPDF: Herwig7 requires two further PDFs
MG5aMC: Herwig7 allows to use it as a ME provider
gengetopt: Herwig7 needs it during compilation
HerwigPP: Removed obsolete patches and a small patch to compile it with our boost version is necessary

### CODE REVIEW

**Comment by bendavid** [comment]
Why are the extra pdfs needed locally, given that they should already be available through cvmfs?

**Comment by mharrend** [comment]
I had to add the following two PDFs to the LHAPDF config since
Herwig7 use them by default for its LO and NLO event generation and
already wants them during make check / make install.
MMHT2014lo68cl MMHT2014nlo68cl
For some reasons, they were not found during my local build using the
cmsdistbuild script.

**Comment by bendavid** [comment]
Ok, indeed I don't think cvmfs access can be guaranteed during the build process, so adding them is necessary in that case.

File: gengetopt.spec [hunk 1]
```diff
@@ -0,0 +1,39 @@
+### RPM external gengetopt 2.22.6
+Source: ftp://ftp.gnu.org/gnu/gengetopt/gengetopt-2.22.6.tar.gz
+Patch0: gengetopt-parallelbuild
+
+BuildRequires: autotools
+
+%define keep_archives true
```
Comment (davidlt) [inline gengetopt.spec]: If you have `--disable-static` why do you `keep_archives true`?

File: gengetopt-toolfile.spec [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+%prep
+
+%build
+
+%install
+
+mkdir -p %i/etc/scram.d
+cat << \EOF_TOOLFILE >%i/etc/scram.d/gengetopt.xml
+<tool name="gengetopt" version="@TOOL_VERSION@">
+  <client>
```
Comment (davidlt) [inline gengetopt-toolfile.spec:12]: Does this package provide any shared library which should be in `LD_LIBRARY_PATH`. If yes, you should add the lines `<lib name="[NAME]"/>` where you have, e.g., `lib[NAME].so`.

File: gengetopt.spec [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+%if "%{?cms_cxx:set}" != "set"
+%define cms_cxx c++
+%endif
+
+%if "%{?cms_cxxflags:set}" != "set"
+%define cms_cxxflags -O2 -std=c++11
```
Comment (davidlt) [inline gengetopt.spec]: You can set this to `c++14` as soonish it will be default for C++.

File: herwigpp.spec [hunk 4]
```diff
@@ -1,29 +1,51 @@
...
+
+BuildRequires: autotools
 
 %if "%{?cms_cxx:set}" != "set"
 %define cms_cxx g++
 %endif
 
 %if "%{?cms_cxxflags:set}" != "set"
-%define cms_cxxflags -O2 -std=c++0x
+%define cms_cxxflags -O2 -std=c++11
```
Comment (davidlt) [inline herwigpp.spec]: Same here, `c++14` as it will be default for C++11 soonish.

File: herwigpp.spec [hunk 5]
```diff
@@ -1,29 +1,51 @@
...
 %endif
 
 %prep
-%setup -q -n herwig++/%{realversion}
+%setup -q -n Herwig-7.0.0
```
Comment (davidlt) [inline herwigpp.spec]: `7.0.0` -> `%{realversion}`

File: herwigpp.spec [hunk 6]
```diff
@@ -1,29 +1,51 @@
-### RPM external herwigpp 2.7.1
-Source: http://service-spi.web.cern.ch/service-spi/external/MCGenerators/distribution/herwig++/herwig++-%{realversion}-src.tgz
-Requires: boost thepeg gsl hepmc
+### RPM external herwigpp 7.0.0
+Source: https://www.hepforge.org/archive/herwig/Herwig-7.0.0.tar.bz2
```
Comment (davidlt) [inline herwigpp.spec]: `7.0.0` -> `%{realversion}`

File: gengetopt.spec (hunk 1 above)
Comment (mharrend) [inline gengetopt.spec]: I was using some other package as a template, so I guess we can remove this line.

File: herwigpp.spec [hunk 7]
```diff
@@ -1,29 +1,51 @@
...
+            --disable-silent-rules \
+            --with-thepeg=$THEPEG_ROOT \
+            --with-fastjet=$FASTJET_ROOT \
+            --with-gsl=$GSL_ROOT \
+            --with-boost=$BOOST_ROOT \
+            --with-madgraph=$MADGRAPH5AMCATNLO_ROOT \
+            --with-openloops=$OPENLOOPS_ROOT \
+            --prefix=%i \
+            CXX="$CXX" CC="$CC" CXXFLAGS="%{cms_cxxflags}" \
+	    BOOST_ROOT="$BOOST_ROOT" LDFLAGS="$LDFLAGS -L$BOOST_ROOT/lib"
```
Comment (davidlt) [inline herwigpp.spec:48]: Why do we need `-L$BOOST_ROOT/lib` in `LDFLAGS`? Is `--with-boost=` not working properly?

File: gengetopt-toolfile.spec (hunk 2 above)
Comment (mharrend) [inline gengetopt-toolfile.spec:12]: No, it does not provide libraries.

File: gengetopt.spec (hunk 3 above)
Comment (mharrend) [inline gengetopt.spec]: Yes, I am fine with this.

File: thepeg.spec [hunk 8]
```diff
@@ -1,17 +1,20 @@
-### RPM external thepeg 1.9.2p1
+### RPM external thepeg 2.0.0
 ## INITENV +PATH LD_LIBRARY_PATH %{i}/lib/ThePEG
 ## INITENV +PATH DYLD_LIBRARY_PATH %{i}/lib/ThePEG
 
-%define tag 41e9a26f5ca9659e30e9c27e4dc86e65ecd4a4bd
-%define branch cms/v%realversion
+# Download from official webpage
+Source: http://www.hepforge.org/archive/thepeg/ThePEG-2.0.0.tar.gz
```
Comment (davidlt) [inline thepeg.spec]: `2.0.0` -> `%{realversion}`

File: thepeg.spec [hunk 9]
```diff
@@ -23,7 +26,7 @@ BuildRequires: autotools
 %endif
 
 %prep
-%setup -q -n thepeg-%{realversion}-%{tag}
+%setup -q -n ThePEG-2.0.0
```
Comment (davidlt) [inline thepeg.spec]: Same here, `2.0.0` -> `%{realversion}`

File: thepeg.spec [hunk 10]
```diff
@@ -58,4 +66,4 @@ find %{i}/lib -name '*.la' -exec rm -f {} \;
 %{relocateConfig}lib/ThePEG/Makefile.common
 %{relocateConfig}lib/ThePEG/Makefile
 %{relocateConfig}lib/ThePEG/ThePEGDefaults.rpo
-%{relocateConfig}lib/ThePEG/ThePEGDefaults-1.9.2.rpo
+%{relocateConfig}lib/ThePEG/ThePEGDefaults-2.0.0.rpo
```
Comment (davidlt) [inline thepeg.spec]: Same here, `2.0.0` -> `%{realversion}`

**Comment by davidlt** [comment]
I would also like to have things divided into separate commits, e.g. gengetopt is fully a stand alone thing. Thus it can go into it's own commit. Things should not be monolithic (one big commit).

gengetopt also can have content in `/usr/share`, make sure you delete it.

File: herwigpp.spec (hunk 7 above)
Comment (mharrend) [inline herwigpp.spec:48]: Yes, we need it and we need also my boost patch. Somehow the m4 boost macro is not working properly.

Comment (davidlt) [inline herwigpp.spec:48]: What exactly goes wrong? If it completely does not detect boost, then you also need `CPPFLAGS='${BOOST_ROOT}/include'` to find the correct headers.

**Comment by mharrend** [comment]
@davidlt Yes, i can split the pull request. 
But can you please explain further what you mean with 
"gengetopt also can have content in /usr/share, make sure you delete it."

However, I would be fine with if you just implement the necessary code lines. I guess it is easier to just implement what you mean instead of explaining it to me first.

Thanks for the comments above I should have included them now.

**Comment by davidlt** [comment]
I looked at gengetopt RPMs and they tend to install documentation in /usr/share directory. At CMS we don't ship man pages, documentations, examples, etc of such tools. We strip the package of not needed files.

**Comment by govoni** [comment]
+1

**Comment by mharrend** [comment]
Closing this request after splitting pull request in two requests:
#2099 Gengetopt request 
#2100 Herwig7 request

**Pull request closed without merging**